e2etests/e2etests.js

```diff
@@ -41,12 +41,16 @@ device_provision(hubConnectionString, function (err, provisionedDevices) {
     console.log('Unable to create the devices needed.');
   } else {
     provisionedDevices.forEach(function(deviceToTest) {
-      acknowledgementProtocols.forEach(function (protocolToTest) {
-        device_acknowledge_tests(hubConnectionString, protocolToTest, deviceToTest);
+      if (deviceToTest.authenticationDescription !== 'shared private key') {
```

Comment:
One didn't have to add these conditions if one didn't mind adding 33% to the time taken to perform this section of the tests.  That's A LOT of time.

I minded.

This is an e2e test program.  Not a public API that we expect some third party to have to consume.  There is NO gain in adding the enumeration.

---
In reply to: [128297732](https://github.com/Azure/azure-iot-sdk-node/pull/61#discussion_r128297732) [](ancestors = 128297732)